src/reporter/github.js

```diff
@@ -25,173 +32,145 @@ export default class GitHub {
+    const [issue] = issues.filter(item => item.title === title); // since only one is expected, use the first one
 
-      return issue;
-    } catch (error) {
-      logger.error(`🤖 Could not find GitHub issue "${title}": ${error}`);
-    }
+    return issue;
   }
 
   async addCommentToIssue({ issue, comment: body }) {
-    try {
-      const { data: comment } = await this.octokit.request('POST /repos/{owner}/{repo}/issues/{issue_number}/comments', {
-        ...this.commonParams,
-        issue_number: issue.number,
-        body,
-      });
+    const { data: comment } = await this.octokit.request('POST /repos/{owner}/{repo}/issues/{issue_number}/comments', {
+      ...this.commonParams,
+      issue_number: issue.number,
+      body,
+    });
 
-      logger.info(`🤖 Added comment to GitHub issue #${issue.number}: ${comment.html_url}`);
-
-      return comment;
-    } catch (error) {
-      logger.error(`🤖 Could not add comment to GitHub issue #${issue.number} "${issue.title}": ${error}`);
-    }
+    return comment;
   }
 
   async closeIssueWithCommentIfExists({ title, comment }) {
-    const openedIssue = await this.getIssue({ title, state: GitHub.ISSUE_STATE_OPEN });
+    try {
+      const openedIssue = await this.getIssue({ title, state: GitHub.ISSUE_STATE_OPEN });
 
-    if (!openedIssue) {
-      return;
-    }
+      if (!openedIssue) {
+        return;
+      }
 
-    await this.addCommentToIssue({ issue: openedIssue, comment });
+      await this.addCommentToIssue({ issue: openedIssue, comment });
+      await this.closeIssue(openedIssue);
 
-    return this.closeIssue(openedIssue);
+      return logger.info(`🤖 Closed issue #${openedIssue.number}: ${openedIssue.html_url}`);
+    } catch (error) {
+      logger.error(`🤖 Failed to handle issue "${title}": ${error.message}`);
+    }
   }
 
   async createOrUpdateIssue({ title, description, label }) {
-    const issue = await this.getIssue({ title, state: GitHub.ISSUE_STATE_ALL });
+    try {
+      const issue = await this.getIssue({ title, state: GitHub.ISSUE_STATE_ALL });
 
-    if (!issue) {
-      return this.createIssue({ title, description, labels: [label] });
-    }
+      if (!issue) {
+        const createdIssue = await this.createIssue({ title, description, labels: [label] });
 
-    if (issue.state == GitHub.ISSUE_STATE_CLOSED) {
-      await this.openIssue(issue);
-    }
+        return logger.info(`🤖 Created GitHub issue #${createdIssue.number} "${title}": ${createdIssue.html_url}`);
+      }
 
-    const managedLabelsNames = this.MANAGED_LABELS.map(label => label.name);
-    const [managedLabel] = issue.labels.filter(label => managedLabelsNames.includes(label.name)); // it is assumed that only one specific reason for failure is possible at a time, making managed labels mutually exclusive
+      if (issue.state == GitHub.ISSUE_STATE_CLOSED) {
+        await this.openIssue(issue);
+        logger.info(`🤖 Reopen issue #${issue.number}: ${issue.html_url}`);
```

Comment:
Do we really need to log this?
```suggestion
        logger.info(`Reopened issue #${issue.number}: ${issue.html_url}`);
```